src/qml/QFieldCloudScreen.qml

```diff
@@ -444,37 +442,65 @@ Page {
-                    font.italic: true
                     color: Theme.secondaryTextColor
                     wrapMode: Text.WordWrap
                   }
                 }
 
-                QfToolButton {
-                  id: menuButton
-                  round: true
-                  opacity: 0.5
+                Item {
                   width: 48
                   height: 48
-
                   anchors.verticalCenter: line.verticalCenter
 
-                  bgcolor: "transparent"
-                  iconSource: Theme.getThemeVectorIcon("ic_dot_menu_black_24dp")
-                  iconColor: Theme.mainTextColor
-
-                  onClicked: mouse => {
-                    let gc = mapToItem(qfieldCloudScreen, 0, 0);
-                    projectActions.projectId = Id;
-                    projectActions.projectOwner = Owner;
-                    projectActions.projectName = Name;
-                    projectActions.projectLocalPath = LocalPath;
-                    downloadProject.visible = LocalPath === '' && Status !== QFieldCloudProject.ProjectStatus.Downloading;
-                    openProject.visible = LocalPath !== '';
-                    viewProjectFolder.visible = LocalPath !== '';
-                    removeProject.visible = LocalPath !== '';
-                    cancelDownloadProject.visible = Status === QFieldCloudProject.ProjectStatus.Downloading;
-                    projectActions.popup(gc.x + width - projectActions.width, gc.y - height);
+                  QfToolButton {
+                    id: downloadActionButton
+
+                    visible: LocalPath === ''
+                    iconSource: Status === QFieldCloudProject.ProjectStatus.Downloading ? Theme.getThemeVectorIcon("ic_clear_white_24dp") : Theme.getThemeVectorIcon("ic_download_white_24dp")
+                    iconColor: Status === QFieldCloudProject.ProjectStatus.Downloading ? Theme.mainTextColor : Theme.mainColor
+                    opacity: Status === QFieldCloudProject.ProjectStatus.Downloading ? .5 : 1
+
+                    onClicked: {
+                      if (Status === QFieldCloudProject.ProjectStatus.Downloading) {
+                        cloudProjectsModel.projectCancelDownload(Id);
+                      } else {
+                        cloudProjectsModel.projectPackageAndDownload(Id);
+                      }
+                    }
+                  }
+
+                  QfToolButton {
+                    id: menuButton
+                    round: true
+                    opacity: 0.5
+                    width: 48
+                    height: 48
+                    visible: LocalPath !== ''
+
+                    bgcolor: "transparent"
+                    iconSource: Theme.getThemeVectorIcon("ic_dot_menu_black_24dp")
+                    iconColor: Theme.mainTextColor
+
+                    onClicked: mouse => {
+                      let gc = mapToItem(qfieldCloudScreen, 0, 0);
```

Comment:
Purely for the sake of easing code reading, can we move this line to sit just above the .popup() call? Also, you can const it.